File: dotnet/niuz.application/dtos/PaymentDto.cs
using System;

namespace niuz.application.dtos
{
    public class PaymentDto
    {
        public int Amount { get; }
        public string BankAccount { get; }
        public string Recipient { get; }
        public string Description { get; }

        public PaymentDto(int amount, string bankAccount, string recipient, string description)
        {
            Amount = amount;
            BankAccount = bankAccount;
            Recipient = recipient;
            Description = description;
        }

        protected bool Equals(PaymentDto other)
        {
            return Amount == other.Amount && BankAccount == other.BankAccount && Recipient == other.Recipient && Description == other.Description;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((PaymentDto) obj);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(Amount, BankAccount, Recipient, Description);
        }
    }
}

File: java/src/main/java/com/dddeurope/niuz/finance/PaymentDto.java
package com.dddeurope.niuz.finance;

import java.util.Objects;

public class PaymentDto {
    private final int amount;
    private final String bankAccount;
    private final String recipient;
    private final String description;

    public PaymentDto(int amount, String bankAccount, String recipient, String description) {
        this.amount = amount;
        this.bankAccount = bankAccount;
        this.recipient = recipient;
        this.description = description;
    }

    @Override
    public String toString() {
        return "PaymentDto{" +
                "amount=" + amount +
                ", bankAccount='" + bankAccount + '\'' +
                ", recipient='" + recipient + '\'' +
                ", description='" + description + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PaymentDto that = (PaymentDto) o;
        return amount == that.amount && Objects.equals(bankAccount, that.bankAccount) && Objects.equals(recipient, that.recipient) && Objects.equals(description, that.description);
    }

    @Override
    public int hashCode() {
        return Objects.hash(amount, bankAccount, recipient, description);
    }
}
